config/src/main/java/com/alibaba/nacos/config/server/auth/PermissionPersistService.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.nacos.config.server.auth;
+
+
+import com.alibaba.nacos.config.server.model.Page;
+import com.alibaba.nacos.config.server.service.PersistService;
+import com.alibaba.nacos.config.server.utils.PaginationHelper;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.jdbc.CannotGetJdbcConnectionException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Service;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+
+import static com.alibaba.nacos.config.server.utils.LogUtil.fatalLog;
+
+/**
+ * Permission CRUD service
+ *
+ * @author nkorange
+ * @since 1.2.0
+ */
+@Service
+public class PermissionPersistService extends PersistService {
+
+    public Page<PermissionInfo> getPermissions(String role, int pageNo, int pageSize) {
+        PaginationHelper<PermissionInfo> helper = new PaginationHelper<>();
+
+        String sqlCountRows = "select count(*) from permissions where ";
```

Comment:
I guess use string builder is better?
And I think inserting 'where' in the front of String variable where instead of append 'where' to the end of string sqlFetchRows is better.